### PR TITLE
Handled formatting invalid module names

### PR DESCRIPTION
**Pull request by scohen**
I'm working in a project that generates invalid module names, which would crash lexical due to lexical using Module.split, which causes a crash when it gets an invalid module name.

Switched to use `String.split` instead.

### CODE REVIEW

File: projects/lexical_shared/lib/lexical/formats.ex [hunk 1]
```diff
@@ -68,8 +68,14 @@ defmodule Lexical.Formats do
...
-      module_name
-      |> Module.split()
+      module_part =
+        case string_name do
+          "Elixir." <> rest -> rest
+          other -> other
+        end
+
+      module_part
+      |> String.split()
```
Comment (zachallaun) [inline projects/lexical_shared/lib/lexical/formats.ex:78]: I'm not sure I'm understanding what this is handling. It's splitting on whitespace, which should never be in the module name, right? So this ends up being a list of one element, and then the join just unwraps the element.

Comment (scohen) [reply projects/lexical_shared/lib/lexical/formats.ex:78]: I'm working in a codebase where module names end up like :"Foo.Bar.Baz", it's very wrong, but legal.

Comment (scohen) [reply projects/lexical_shared/lib/lexical/formats.ex:78]: Oh, never mind, I got ya.

File: projects/lexical_shared/test/lexical/formats_test.exs [hunk 2]
```diff
@@ -15,6 +15,14 @@ defmodule Lexical.FormatsTest do
     test "it correctly handles an erlang module name" do
       assert ":ets" == Formats.module(:ets)
     end
+
+    test "it drops any `Elixir.` prefix" do
+      assert "Kernel.SpecialForms" == Formats.module(Elixir.Kernel.SpecialForms)
+    end
```
Comment (scohen) [inline projects/lexical_shared/test/lexical/formats_test.exs:21]: FYI, @Moosieus this this is basically the same thing as the above tests `Elixir.Kernel.SpecialForms` -> `Kernel.SpecialForms`

**Review by Moosieus — APPROVED** [review]

**Pull request merged**